qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AllergiesSteps.java

```diff
@@ -17,46 +17,63 @@
 import io.cucumber.java.en.And;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
 
 import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.AddOrEditAllergyPage;
 import org.openmrs.contrib.qaframework.page.AllergyPage;
+import org.openmrs.contrib.qaframework.page.FindPatientPage;
 
 public class AllergiesSteps extends Steps {
 	
+	private static final String DRUG_NAME = "Morphine";
+	private static final String REACTION = "Itching";
+	private static final String NEW_REACTION = "Rash";
+	private static final String ALLERGY_NOTE = "The whole body itches";
+	private static final String NEW_ALLERGY_NOTE = "There's a rash on the body";
 	private AllergyPage allergyPage;
+	private AddOrEditAllergyPage addOrEditAllergyPage;
+	private TestData.PatientInfo testPatient;
 
-	@Before(RunTest.HOOK.SELENIUM_DASHBOARD)
+	@Before(RunTest.HOOK.SELENIUM_ALLERGIES)
 	public void visitDashboard() {
-		initiatePatientDashboard();
+		initiateWithLogin();
+		testPatient = createTestPatient();
+		findPatientPage = (FindPatientPage) homePage.goToFindPatientRecord().waitForPage();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();	
 	}
 
-	@After(RunTest.HOOK.SELENIUM_DASHBOARD)
+	@After(RunTest.HOOK.SELENIUM_ALLERGIES)
 	public void destroy() {
+		deletePatient(testPatient);
 		quit();
 	}
 
-	@Given("a user clicks on Allergies link from Patient dashboard")
-	public void loadAllergiesPage() {
+	@Given("a user clicks on Allergies link from the Patient dashboard")
+	public void loadAllergiesBoardPage() {
```

Comment:
i would still maintain `loadAllergiesPage`

Reply:
@sherrif10 I have attended to the review.